src/web/browser/sentryLoader/init.ts

```diff
@@ -0,0 +1,77 @@
+import '../webpackPublicPath';
+import { startup } from '@root/src/web/browser/startup';
+
+import { isAdBlockInUse } from './detectAdBlocker';
+
+const init = () => {
+    // Sentry lets you configure sampleRate to reduce the volume of events sent
+    // but this filter only happens _after_ the library is loaded. The Guardian
+    // measures page views in the billions so we only want to log 1% of errors that
+    // happen but if we used sampleRate to do this we'd be needlessly downloading
+    // Sentry 99% of the time. So instead we just do some basic math here
+    // and use that to prevent the Sentry script from ever loading.
+    const randomNumber = Math.floor(Math.random() * 100) + 1;
+    if (randomNumber !== 42) return Promise.resolve(); // 42 is arbitrary
+
+    return Promise.resolve().then(() => {
```

Comment:
Just out of interest – is there a reason to use `Promise.resolve().then` rather than `new Promise` here? Possibly to defer evaluation a tick?

Reply:
You’d have to change the `return`s to `resolve` If you’re going with `new Promise((resolve, reject)=> {})`

Reply:
I paired on this with @jonathonherbert and we made a change where we simply mark the `init()` as async so there's no need to explicitly return promises anymore.

It took a while before I was able to mangle my mental model of how async / await works into something more aligned with reality but the result is the code is a lot simpler and I now understand `async` a lot better!